### PR TITLE
1364 - Tree enhancements

### PR DESCRIPTION
**Pull request by mdevale**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This pull request solves the problems mentioned in github issue: #1364

**Related github/jira issue (required)**:
Closes: #1364

**Steps necessary to review your pull request (required)**:
Pull this branch.
Run following pages with respect to the problems mentioned in github issue: #1364
1. http://localhost:4000/components/tree/test-dropdown.html
1. http://localhost:4000/components/tree/test-add-node-inbetween-node.html
1. http://localhost:4000/components/tree/test-checkbox-particular-node.html
1. http://localhost:4000/components/tree/test-custom-icon.html

### CODE REVIEW

**Comment by tmcconechy** [comment]
Please merge master on and it should fix your tests 👍

**Comment by mdevale** [comment]
Hello,

Yes will do.

Thanks,
Madhukar

File: src/components/tree/tree.js [hunk 1]
```diff
@@ -140,7 +140,8 @@ Tree.prototype = {
    * @param {object} a an anchor tag reference wrapped in a jQuery object.
    * @returns {void}
```
Comment (tmcconechy) [inline src/components/tree/tree.js]: Could add param docs here

File: src/components/tree/tree.js [hunk 2]
```diff
@@ -1525,7 +1546,13 @@ Tree.prototype = {
    * @param {object} location in tree.
    * @returns {object} li added
    */
-  addNode(nodeData, location) {
+
+  // Parameter added - to add node before or after the node
+  addNode(nodeData, location, isBeforeOrAfter) {
```
Comment (tmcconechy) [inline src/components/tree/tree.js]: Add docs please

**Comment by tmcconechy** [comment]
@mdevale Can you please check my comments so we can update this PR and merge soon. Thanks!

File: src/components/tree/tree.js (hunk 2 above)
Comment (mdevale) [reply src/components/tree/tree.js]: http://localhost:4000/components/tree/test-dropdown.html
I'm not seeing any dropdown?
- Now you will able to see dropdown

http://localhost:4000/components/tree/test-add-node-inbetween-node.html
how do i test this one?
- Click on Add Parent Node and Expand any node and click on Add Leaf Node

http://localhost:4000/components/tree/test-checkbox-particular-node.html
is it supposed to show no checkboxes at all?
- No. Show checkbox for particular node.

**Comment by tmcconechy** [comment]
But one more suggestion is to add e2e tests for these so in the future we will know if anything breaks. To do this add a few tests like the ones in: https://github.com/infor-design/enterprise/blob/master/test/components/tree/tree.e2e-spec.js

**Pull request merged**